### PR TITLE
[eclipse/xtext#1679]ported test code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1679]ported test code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/TestLanguageIdeContentProposalProvider.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+public class TestLanguageIdeContentProposalProvider extends IdeContentProposalProvider {
+	@Inject
+	private TestLanguageGrammarAccess testLanguageGrammarAccess;
+
+	@Override
+	protected void _createProposals(RuleCall ruleCall, ContentAssistContext context,
+			IIdeContentProposalAcceptor acceptor) {
+		boolean _equals = testLanguageGrammarAccess.getTypeDeclarationRule().equals(ruleCall.getRule());
+		if (_equals) {
+			IdeContentProposalCreator _proposalCreator = this.getProposalCreator();
```
Comment (nbhusare) [inline org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/TestLanguageIdeContentProposalProvider.java:33]: `this.getProposalCreator();` -> `getProposalCreator();`

File: org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+/**
+ * @author Sven Efftinge - Initial contribution and API
+ */
+public class CodeActionService implements ICodeActionService2 {
+	@Inject
+	private IChangeSerializer serializer;
+
+	@Override
+	public List<Either<Command, CodeAction>> getCodeActions(ICodeActionService2.Options options) {
+		ArrayList<Either<Command, CodeAction>> actions = new ArrayList<>();
```
Comment (nbhusare) [inline org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.java:50]: `ArrayList<Either<Command, CodeAction>` -> `List<Either<Command, CodeAction>`

File: org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+		textEdit.setRange(d.getRange());
+		workspaceEdit.getChanges().put(options.getCodeActionParams().getTextDocument().getUri(),
+				Lists.newArrayList(textEdit));
+		command.setArguments(Lists.newArrayList(workspaceEdit));
+		return command;
+	}
+
+	private CodeAction fixUnsortedMembers(Diagnostic d, ICodeActionService2.Options options) {
+		WorkspaceEdit wsEdit = recordWorkspaceEdit(options, (Resource copiedResource) -> {
+			Model model = Iterables.getFirst(Iterables.filter(copiedResource.getContents(), Model.class), null);
```
Comment (nbhusare) [inline org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.java:79]: `model` can be `null`.

File: org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		codeAction.setEdit(wsEdit);
+		return codeAction;
+	}
+
+	private WorkspaceEdit recordWorkspaceEdit(ICodeActionService2.Options options,
+			IChangeSerializer.IModification<Resource> mod) {
+		ResourceSet rs = options.getLanguageServerAccess().newLiveScopeResourceSet(options.getResource().getURI());
+		Resource copy = rs.getResource(options.getResource().getURI(), true);
+		serializer.addModification(copy, mod);
+		ArrayList<IEmfResourceChange> documentchanges = new ArrayList<>();
```
Comment (nbhusare) [inline org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/server/CodeActionService.java:96]: `ArrayList<IEmfResourceChange>` -> `List<IEmfResourceChange>`

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/parameters/AbstractParametersParsingTest.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+	}
+
+	@Override
+	protected boolean shouldTestSerializer(XtextResource resource) {
+		return false;
+	}
+
+	@Override
+	public ParserRuleParameters getModel(String model) throws Exception {
+		EObject _model = super.getModel(model);
```
Comment (nbhusare) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/parameters/AbstractParametersParsingTest.java:38]: `_model` -> `model`  just to be consistent with rest of the code

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/parameters/AbstractParametersParsingTest.java [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+		return ((ParserRuleParameters) _model);
```
Comment (nbhusare) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/parameters/AbstractParametersParsingTest.java:39]: Extra `((`

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/parameters/AbstractParametersParsingTest.java [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+	}
+
+	protected ParserRuleParameters getModelWithErrors(String model) throws Exception {
+		EObject _modelAndExpect = getModelAndExpect(model, AbstractXtextTests.EXPECT_ERRORS);
```
Comment (nbhusare) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/parameters/AbstractParametersParsingTest.java:43]: `_modelAndExpect` -> `modelAndExpect`

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/parameters/ParametersParsingTest.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.parser.parameters;
+
+/**
+ * @author Sebastian Zarnekow - Initial contribution and API
+ */
+public class ParametersParsingTest extends AbstractParametersParsingTest {
+	private ParametersTestLanguageStandaloneSetup _parametersTestLanguageStandaloneSetup;
```
Comment (nbhusare) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/parameters/ParametersParsingTest.java:15]: `_parametersTestLanguageStandaloneSetup` -> `parametersTestLanguageStandaloneSetup` for consistency

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/parameters/ParametersParsingTest.java [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+		_parametersTestLanguageStandaloneSetup = new ParametersTestLanguageStandaloneSetup();
```
Comment (nbhusare) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/parser/parameters/ParametersParsingTest.java:20]: `_parametersTestLanguageStandaloneSetup` -> `parametersTestLanguageStandaloneSetup`

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/ResourceSetTest.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+import java.util.HashMap;
+
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+
+public class ResourceSetTest extends AbstractResourceSetTest {
+	@Override
+	protected ResourceSetImpl createEmptyResourceSet() {
+		ResourceSetImpl rs = new ResourceSetImpl();
+		rs.setURIResourceMap(new HashMap<>());
```
Comment (nbhusare) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/ResourceSetTest.java:19]: You might like to use `Collections.emptyMap()` here.

Comment (cdietrich) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/ResourceSetTest.java:19]: want to keep semantics

Comment (szarnekow) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/resource/ResourceSetTest.java:19]: The resourceSet will later populate the map. `#emptyMap()` will fail on that attempt.

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/Issue1161Test.java [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+
+/**
+ * @author Lorenzo Addazi - Initial contribution and API
+ */
+@RunWith(XtextRunner.class)
+@InjectWith(XtextInjectorProvider.class)
```
Comment (szarnekow) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/Issue1161Test.java:37]: Interesting ... custom Setup + InjectorProvider ...

**Pull request merged**